test/pkg/environment/aws/expectations.go

```diff
@@ -525,3 +527,35 @@ func ignoreAlreadyContainsRole(err error) error {
 	}
 	return err
 }
+
+func ExpectCapacityReservationCreated(
+	ctx context.Context,
+	ec2api *ec2.Client,
+	instanceType ec2types.InstanceType,
+	zone string,
+	capacity int32,
+	endDate *time.Time,
+	tags map[string]string,
+) (id string, cleanup func()) {
+	GinkgoHelper()
+	out, err := ec2api.CreateCapacityReservation(ctx, &ec2.CreateCapacityReservationInput{
+		InstanceCount:         lo.ToPtr(capacity),
+		InstanceType:          lo.ToPtr(string(instanceType)),
+		InstancePlatform:      ec2types.CapacityReservationInstancePlatformLinuxUnix,
+		AvailabilityZone:      lo.ToPtr(zone),
+		EndDate:               endDate,
+		InstanceMatchCriteria: ec2types.InstanceMatchCriteriaTargeted,
+		TagSpecifications: lo.Ternary(len(tags) != 0, []ec2types.TagSpecification{{
+			ResourceType: ec2types.ResourceTypeCapacityReservation,
+			Tags:         utils.MergeTags(tags),
+		}}, nil),
+	})
+	Expect(err).ToNot(HaveOccurred())
+	return lo.FromPtr(out.CapacityReservation.CapacityReservationId), func() {
```

Comment:
To me, it's a little odd to return the cleanup function in an expectation like this -- I'd just create a separate cleanup function and call DeferCleanup, cleanup could even take in multiple reservation ids so that you just store ids rather than storing cleanup functions